.github/workflows/docs.yml

```diff
@@ -4,20 +4,39 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
 permissions:
   contents: write
 jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v5
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'  # Pin for consistency
+    - name: Install Dependencies
+      run: ./pw uv sync --only-group docs
+    - name: Build Documentation
+      run: ./pw uv run mkdocs build --strict
   deploy:
+    needs: build
+    # Only deploy docs on pushes to main branch
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v5
       - name: Configure Git Credentials
         run: |
           git config user.name github-actions[bot]
           git config user.email 41898282+github-actions[bot]@users.noreply.github.com
       - uses: actions/setup-python@v5
         with:
-          python-version: 3.x
+          python-version: '3.12'
```

Comment:
to avoid duplicating the version number and make it easier to update in the future, can we make this a variable at the top of the file